### PR TITLE
partitioned-table: remove some restrictions on global index

### PR DESCRIPTION
**Pull request by Defined2014**
### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.5 (TiDB 8.5 versions)
- [ ] v8.4 (TiDB 8.4 versions)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/19382
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: partitioned-table.md [hunk 1]
```diff
@@ -1676,7 +1691,7 @@ ERROR 8264 (HY000): Global Index is needed for index 'a', since the unique index
 
 为解决这些问题，TiDB 从 v8.3.0 开始引入全局索引。全局索引能覆盖整个表的数据，使得主键和唯一键在不包含分区键的情况下仍能保持全局唯一性。同时，全局索引可以在一次操作中访问多个分区的数据，显著提升了针对非分区键的查询性能。
 
-如果你需要创建的唯一索引**不包含分区表达式中使用的所有列**，可以通过在索引定义中添加 `GLOBAL` 关键字来实现。
+如果你需要创建全局索引可以通过在索引定义中添加 `GLOBAL` 关键字来实现。
```
Comment (hfxsd) [inline partitioned-table.md:1694]:
```suggestion
如果你需要创建全局索引，可以通过在索引定义中添加 `GLOBAL` 关键字来实现。
```

**Review by Frank945946 — APPROVED** [review]

File: partitioned-table.md [hunk 2]
```diff
@@ -1676,7 +1691,7 @@ ERROR 8264 (HY000): Global Index is needed for index 'a', since the unique index
 
 为解决这些问题，TiDB 从 v8.3.0 开始引入全局索引。全局索引能覆盖整个表的数据，使得主键和唯一键在不包含分区键的情况下仍能保持全局唯一性。同时，全局索引可以在一次操作中访问多个分区的数据，显著提升了针对非分区键的查询性能。
 
-如果你需要创建的唯一索引**不包含分区表达式中使用的所有列**，可以通过在索引定义中添加 `GLOBAL` 关键字来实现。
+如果你需要创建全局索引，可以通过在索引定义中添加 `GLOBAL` 关键字来实现。
```
Comment (lilin90) [inline partitioned-table.md:1694]:
```suggestion
如果你需要为主键或唯一键创建全局索引，可以通过在索引定义中添加 `GLOBAL` 关键字来实现。
```

**Review by lilin90 — APPROVED** [review]
Rest LGTM

**Pull request merged**